src/pages/ExamsPage/components/AllowanceListActions.jsx

```diff
@@ -10,8 +11,106 @@ import {
 import { DeleteOutline, EditOutline } from '@openedx/paragon/icons';
 import { useIntl } from '@edx/frontend-platform/i18n';
 
+import React, { useState } from 'react';
 import messages from '../messages';
-import { useDeleteAllowance } from '../hooks';
+import { useDeleteAllowance, useEditAllowance, useAllowancesData } from '../hooks';
+
+const EditModal = (isOpen, onClose, allowance, formatMessage) => {
+  const editAllowance = useEditAllowance();
+  const {
+    allowancesList,
+  } = useAllowancesData();
+
+  const allowances = allowancesList.filter(a => a.id === allowance.id);
+  const [additionalTimeError, setAdditionalTimeError] = useState(false);
+
+  const initialFormState = {};
+  const [form, setForm] = useState(initialFormState);
+
+  const handleChange = (event) => {
+    const { name, value } = event.target;
+    setForm(prev => ({ ...prev, [name]: value }));
+  };
+
+  const onSubmit = () => {
+    // todo: make sure that validation is correct
+    setAdditionalTimeError(!form['extra-time-mins']);
+    const valid = (
+      form['extra-time-mins']
+    );
+    if (valid) {
+      form['extra-time-mins'] = Number(form['extra-time-mins']);
```

Comment:
TODO: I'm wondering if this type change should happen in the validation step or earlier to make sure this is valid.

Reply:
Maybe you could cast it as:
```javascript
const extraTimeMins  = +form['extra-time-mins'] || 0;
```
and then check the validation as `extraTimeMins > 0`.

Are we showing an error to the user if validation fails?